File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        if len(height)<2:
            return 0

        l,r=0,len(height)-1
        max_water=0
        
        while l<r:
            
            max_water=max(max_water,(r-l)*(min(height[l],height[r])))
            print(min(height[l],height[r]))
            if height[r]>height[l]:
                l+=1
            elif height[r]<height[l]:
                r-=1
            else:
                r-=1
                l+=1
        return max_water

        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxWaterContainer=0;
        while(left<right){
            int minHeight=Math.min(height[left],height[right]);
            maxWaterContainer=Math.max(maxWaterContainer,minHeight*(right-left));
            if(height[left]>height[right]){
                right--;
            }
            else{
                left++;
            }
        }
        return maxWaterContainer;
        
    }
}